src/Features/CSharp/Portable/CodeFixes/RemoveNewModifier/RemoveNewModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,131 @@
+            => WellKnownFixAllProviders.BatchFixer;
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(CS0109);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            var token = root.FindToken(diagnosticSpan.Start);
+
+            var memberDeclarationSyntax = token.GetAncestor<MemberDeclarationSyntax>();
+            if (memberDeclarationSyntax == null)
+                return;
+
+            var newModifier = GetNewModifier(memberDeclarationSyntax, CSharpSyntaxFacts.Instance);
+            if (newModifier == default)
+                return;
+
+            context.RegisterCodeFix(
+                new MyCodeAction(ct => FixAsync(context.Document, memberDeclarationSyntax, ct)),
+                context.Diagnostics);
+        }
+
+        private async Task<Document> FixAsync(Document document, MemberDeclarationSyntax node, CancellationToken cancellationToken)
+        {
+            var syntaxFacts = CSharpSyntaxFacts.Instance;
+
+            var newModifier = GetNewModifier(node, syntaxFacts);
+
+            var newNode = node;
+
+            if (newModifier.HasTrailingTrivia || newModifier.HasLeadingTrivia)
+            {
+                var newModifierTrivia = newModifier.GetAllTrivia().ToSyntaxTriviaList();
+                var previousToken = newModifier.GetPreviousToken();
+                var nextToken = newModifier.GetNextToken();
+
+                var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                var isFirstTokenOnLine = newModifier.IsFirstTokenOnLine(sourceText);
+
+                var newTrivia = new SyntaxTriviaList();
+                if (!isFirstTokenOnLine)
+                    newTrivia = newTrivia.AddRange(previousToken.TrailingTrivia);
+                newTrivia = newTrivia
+                    .AddRange(newModifierTrivia)
+                    .AddRange(nextToken.LeadingTrivia);
+                newTrivia = CollapseSequentialWhitespaceTrivia(newTrivia);
+
+                if (isFirstTokenOnLine)
+                {
+                    var nextTokenWithMovedTrivia = nextToken.WithLeadingTrivia(newTrivia);
+                    newNode = newNode.ReplaceToken(nextToken, nextTokenWithMovedTrivia);
+                }
+                else
+                {
+                    var previousTokenWithMovedTrivia = previousToken.WithTrailingTrivia(newTrivia);
+                    newNode = newNode.ReplaceToken(previousToken, previousTokenWithMovedTrivia);
+                }
+            }
+
+            newNode = newNode.ReplaceToken(GetNewModifier(newNode, syntaxFacts), SyntaxFactory.Token(SyntaxKind.None));
```

Comment:
instead of all the replace-tokens, consider just making a SyntaxEditor.  Then perform two edits on it similar to this:

```c#
editor.ReplaceNode(
  nextToken.Parent,
  nextToken.Parent.ReplaceToken(nextToken, nextTokenWithMovedTrivia));
editor.ReplaceNode(
  memberDecl,
  (current, _) =>
  {
    var currentMemberDecl = (MemberDeclarationSyntax)current;
    return currentMemberDecl.WithModifiers(RemoveNew(currentMemberDecl.Modifiers));
  });
```

Reply:
I'm receiving `current` as `null` when calling `ReplaceNode` with callback: https://github.com/vanashimko/roslyn/commit/067050ef86fd34ab150b22b420163d20dbcd49ff. Seems to be I'm doing something wrong again :(